README.md

```diff
@@ -158,6 +158,16 @@ az containerapp update \
 * Note: The image must be pushed to [ACR](https://azure.microsoft.com/services/container-registry/) or other registry service.
 See [official docs](https://docs.microsoft.com/azure/container-apps/) for more information.
 
+Or [Oracle Cloud Functions](https://www.oracle.com/cloud-native/functions/)
+
+```
+fn update function \
+  my-container-app my-function-name \
+  --image $(ko build --bare -P . | sed 's/@sha256:/:sha256-/')
```

Comment:
Again, I'd like to avoid recommending that we reference the image by tag, for a few reasons:

1. This is not generally considered a best practice.
2. Their docs don't seem to indicate that it's required anyway. If it _is_ required to deploy by tag, I'd prefer not to recommend using it for that reason.
3. The image won't be pushed with that tag, so deploying will fail using these instructions.

Also, `--bare` and `-P` specify conflicting behavior. It should be an error to specify both, there's a bug for that somewhere. If `--bare` is required to make this work, we should add that in a `* Note` below.

We can also remove the note that OCIR is required, but we should keep the link to official docs.

Reply:
>We can also remove the note that OCIR is required, but we should keep the link to official docs.

Current note is the same as that of Azure. Do you want to remove?

Reply:
How about this?

>* Note: The image must be pushed to [OCIR](https://www.oracle.com/cloud-native/container-registry/) or other registry service. You need to specify `--bare` since OCI does not support image by digest. See [official docs](https://docs.oracle.com/en-us/iaas/Content/Functions/Tasks/functionscreatefncontext.htm) for more information.

Reply:
What I mean is, the docs seem to indicate that OCIR isn't required. We should check whether that's the case before stating that it is required. It also sounds like OCF _does_ support images by digest, so we should check that too before saying it doesn't.

Reply:
Okay. Could you please specify the command you want to try? (Will do that tomorrow, it is currently midnight)

Reply:
```
fn update function \
  my-container-app my-function-name \
  --image $(ko build ./)
```

If that doesn't work (you had a 400 before), try `--bare`. You may also want to try another registry besides OCIR, such as DockerHub or GHCR, where the image is publicly accessible. That would tell us whether OCF actually requires OCIR, or if that's just what's documented in the docs.

Thanks for sticking with this, I really appreciate the contribution. I just want to make sure it's as generally applicable and useful to anybody who finds it, and it doesn't help that I'm personally so unfamiliar with Oracle Cloud.

Reply:
Unfortunately, OCF does not support external registry. When creating new function, registry and image is not possible to be registered from external. It must be provided on OCIR.

![image](https://user-images.githubusercontent.com/10111/160027183-70ea7ea8-d22a-4e0b-954e-6c1884a21b1c.png)

![image](https://user-images.githubusercontent.com/10111/160027233-d996b0e0-7f46-470b-9c58-d2b52729610a.png)

![image](https://user-images.githubusercontent.com/10111/160027339-fc90ba70-cb7a-4b13-8348-a6d9118474ea.png)

And also, fn cli does not treat non-OCIR image.

>Fn: Service error:InvalidParameter. Invalid Invalid image - The image must be an OCIR image in this region's registry. http status code: 400. Opc request

Reply:
Additional: fn cli or OCF does not handle image by digest. So

```
fn update function \
  my-container-app my-function-name \
  --image $(ko build ./)
```
Did not work.

Reply:
Thanks for testing that out. What was the error when you tried to deploy by digest? It might be possible that deploying by digest is allowed, but that OCIF doesn't accept image names like `ko` produces by default -- some registries don't.

If OCF _doesn't_ support deploying functions from images by digest, I'm inclined not to recommend its use, since this is generally considered a best practice.